dask/dataframe/groupby.py

```diff
@@ -132,6 +137,340 @@ def _nunique_series_aggregate(df):
     return df.groupby(df.columns[1])[df.columns[0]].nunique()
 
 
+###############################################################
+# Aggregate support
+#
+# Aggregate is implemented as:
+#
+# 1. group-by-aggregate all partitions into intermediate values
+# 2. collect all partitions into a single partition
+# 3. group-by-aggregate the result into intermediate values
+# 4. transform all intermediate values into the result
+#
+# In Step 1 and 3 the dataframe is grouped on the same columns.
+#
+###############################################################
+def _make_agg_id(func, column):
+    return '{!s}-{!s}-{}'.format(func, column, tokenize(func, column))
```

Comment:
Are func and column always strings?  If so then perhaps just `func + '-' + column`?